src/main/java/picard/sam/SamToFastq.java

```diff
@@ -165,10 +165,13 @@ public static void main(final String[] argv) {
     protected int doWork() {
         IOUtil.assertFileIsReadable(INPUT);
         final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
-        final Map<String, SAMRecord> firstSeenMates = new HashMap<String, SAMRecord>();
+        final Map<String, SAMRecord> firstSeenMates = new HashMap<>();
         final FastqWriterFactory factory = new FastqWriterFactory();
         factory.setCreateMd5(CREATE_MD5_FILE);
         final Map<SAMReadGroupRecord, FastqWriters> writers = generateWriters(reader.getFileHeader().getReadGroups(), factory);
+        if (writers.isEmpty()) {

```

Comment:
The error message should be more informative here.
Given that this tool has the option to emit per-readgroup fastqs _or_ fastqs ignoring the readgroup, the error message should say something to the effect of "consider not using the 'OUTPUT_PER_RG' option" 

if the OUTPUT_PER_RG was not used, this is a bug that needs further investigation.....